mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_workload.go

```diff
@@ -0,0 +1,188 @@
+                                                                    Computed: true,
+                                                                },
+                                                                "location":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                                "zone":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                            },
+                                                        },        
+                                                },
+                                        },
+                                },
+                        },
+        },
+    }
+}
+
+func dataSourceApphubDiscoveredWorkloadRead(d *schema.ResourceData, meta interface{}) error {
+        config := meta.(*transport_tpg.Config)
+        userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+        if err != nil {
+                return err
+        }
+
+        url, err := tpgresource.ReplaceVars(d, config, "{{ApphubBasePath}}projects/{{project}}/locations/{{location}}/discoveredWorkloads:lookup?uri={{workload_uri}}" )
+        if err != nil {
+                return err
+        }
+
+        billingProject := ""
+
+        // err == nil indicates that the billing_project value was found
+        if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+                billingProject = bp
+        }
+        
+        var res map[string]interface{}
+
+        err = transport_tpg.Retry(transport_tpg.RetryOptions{
+        RetryFunc: func() (rerr error) {
+            res, rerr = transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+                Config:    config,
+                Method:    "GET",
+                Project:   billingProject,
+                RawURL:    url,
+                UserAgent: userAgent,
+            })
+            return rerr
+            },
+            Timeout: d.Timeout(schema.TimeoutRead),
+        })
+
+        if err != nil {
+            return transport_tpg.HandleDataSourceNotFoundError(err, d, fmt.Sprintf("ApphubDiscoveredWorkload %q", d.Id()), url)
+        }
+        
+        if err := d.Set("discovered_workload", flattenApphubDiscoveredWorkload(res["discoveredWorkload"], d, config)); err != nil {
+                return fmt.Errorf("Error setting discovered workload: %s", err)
+        }
+        
+        id, err := tpgresource.ReplaceVars(d, config, "{{workload_uri}}")
```

Comment:
Sorry, after reading the documentation, it looks like the id should be the name of the workload.

Reply:
Done.